guides/source/action_controller_overview.md

```diff
@@ -899,6 +899,61 @@ Rails collects all of the parameters sent along with the request in the `params`
 [`query_parameters`]: https://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-query_parameters
 [`request_parameters`]: https://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-request_parameters
 
+#### `request.variant`
+
+Controllers might need to tailor a response based on context-specific information in a request. For example, controllers responding to requests from a mobile platform might need to render a HTML, JSON, or XML different content than requests from a desktop browser. They can add that context to the requests variant so that views render appropriately. Variant names are arbitrary, and can communicate anything from the request's platform (`:anrdoid`, `:ios`, `:linux`, `:macos`, `:windows`) to its browser (`:chrome`, `:edge`, `:firefox`, `:safari`), to the type of user (`:admin`, `:guest`, `:user`).
+
+You can set the [`request.variant`](https://api.rubyonrails.org/classes/ActionDispatch/Http/MimeNegotiation.html#method-i-variant-3D) in a `before_action`:
+
+```ruby
+request.variant = :tablet if request.user_agent.include?("iPad")
+```
+
+Responding with a variant in a controller action is like responding with a format:
+
+```ruby
+# app/controllers/projects_controller.rb
+
+def show
+  # ...
+  respond_to do |format|
+    format.html do |html|
+      html.tablet                         # renders app/views/projects/show.html+tablet.erb
+      html.phone { extra_setup; render }  # renders app/views/projects/show.html+phone.erb
+    end
+  end
+end
+```
+
+Provide separate templates for each format and variant:
+
+* `app/views/projects/show.html.erb`
+* `app/views/projects/show.html+tablet.erb`
+* `app/views/projects/show.html+phone.erb`
+
+You can also simplify the variants definition using the inline syntax:
+
+```ruby
+respond_to do |format|
+  format.html.tablet
+  format.html.phone  { extra_setup; render }
+end
+```
+
+The [`request.variant`](https://api.rubyonrails.org/classes/ActionDispatch/Http/MimeNegotiation.html#method-i-variant) method returns an instance of [`ActiveSupport::ArrayInquirer`](https://api.rubyonrails.org/classes/ActiveSupport/ArrayInquirer.html):
+
+```ruby
+request.variant = :phone
+request.variant.phone?  # => true
+request.variant.tablet? # => false
+
+request.variant = [:phone, :tablet]
+request.variant.phone?                  # => true
+request.variant.desktop?                # => false
+request.variant.any?(:phone, :desktop)  # => true
+request.variant.any?(:desktop, :watch)  # => false
+```
```

Comment:
I would leave this bit out. I think the controller example is a good way to introduce the concept of variants from a guide standpoint but the `ArrayInquirer` detail feels like it belongs in the API doc.

Reply:
My intention was to remove everything highlighted here, do you think we should keep the predicate API?